packages/forms/src/index.tsx

```diff
@@ -113,6 +118,59 @@ const inputTagProps = <T extends InputTagProps>(
   return tagProps
 }
 
+// A hook-like function merge field props and return props (ref, onChange and onBlur primarily) of RHF v7 register()
+const useFieldRegister = <
+  T extends Omit<ValidatableFieldProps, 'defaultValue'>,
+  E
+>(
+  props: T,
+  ref?: React.ForwardedRef<E>
+) => {
+  const { register } = useFormContext()
+  const validation = props.validation || { required: false }
+
+  // Primarily for TextAreaField
+  if (!validation.validate && props.transformValue === 'Json') {
+    validation.validate = jsonValidation
+  }
```

Comment:
This logic mainly for `TextAreaField` but it would be cleaner to encapsulate in this hook-like function, any concerns?

Reply:
This seems ok since, even if the field isn't a text area, it's providing default validation for fields expecting json